### ISSUE TITLE
Make installing binary releases a native experience

### ISSUE THREAD

**Issue by leighmcculloch** [stellar/stellar-cli#1446]
Installing the CLI from source is the primary way we advertise installing the CLI.

However, building from source is time consuming and makes some assumptions about environment setup, additional compilers like a C compiler, and uses non-trivial amounts of memory resources, bandwidth, CPU capabilities, and ... patience.

We also prebuild binaries for Windows (x86), macOS (x86, arm64), and Linux (x86, arm64). Theoretically no one needs to build from source except those on esoteric systems. In terms of commands on the command line installing from source is the simplest, and so I think it is highly unlikely that folks use the prebuilt binaries much, especially given than our CI job for publishing the binaries is partly broken and requires manual running after each release and even on times we've forgotten no one has complained.

We should make these binaries easy to install using tools folks already have on their system.

_While we do support cargo-bininstall, that binary is not often installed for folks and itself is a reasonably large installation. It's a good option to have on hand for folks who have it, but shouldn't be the core of the distribution story._

I think we should make a one line shell script install that works for the majority of users:

```
curl -L https://cli.stellar.org | bash
```

And similar for Windows if such a command exists, or `winget` is probably more appropriate given it is built-in to Windows.

I think we should also add release targets for all of the following, over time. For any platform where multiple options are listed, I've ordered them by priority.

### Crossplatform
- [ ] ~curl | bash (e.g. `curl -L https://cli.stellar.org | bash`)~ __[Edit: Deprioritizing for now.]__
- [ ] ~NPM, that installs the prebuilt binary (e.g. `npm -g install @stellar/stellar-cli`)~ __[Edit: Deprioritizing for now.]__
- [ ] ~Cargo, that installs the prebuilt binary (e.g. `cargo install stellar-cli-bin`)~ __[Edit: Deprioritizing for now.]__
- [ ] ~Docker (e.g. `docker run -i stellar/stellar-cli`)~ https://github.com/stellar/stellar-cli/pull/1550

### Windows (x86)
- [ ] Winget (e.g. `winget install Stellar.StellarCLI`) @fnando 

### macOS (x86, arm64)
- [x] Homebrew - We currently run a homebrew source-only tap. https://github.com/Homebrew/homebrew-core/pull/180191 @leighmcculloch 

### Linux (x86, arm64)
- [x] Homebrew - Same as above for supporting Homebrew on Linux. https://github.com/Homebrew/homebrew-core/pull/180191
- [ ] ~Debian/Ubuntu~
    - ~SDF hosts an apt repo that could house the stellar-cli as a package to install.~

### CI
- [x] GitHub Actions - Add an action to the repo so the cli can be installed from binary with: @leighmcculloch 
    `- uses: stellar/stellar-cli@v21.0.0` @leighmcculloch

**Comment by leighmcculloch** [comment]
Of the options the most important of these to prioritise is probably `winget`, because it seems like of folks who have issues building and installing the CLI, windows users are most represented. I'd happily take it, but I don't have access to windows and it'd be good if a native windows user could explore it. (cc @fnando Not sure if that's something you could do with your windows machine.)

**Comment by leighmcculloch** [comment]
Of the options that are low hanging fruit quick wins, that's `homebrew` binaries, because we already have a homebrew-tap and it's a short step to add binaries 🤞🏻. I'm going to take doing that because I'm familiar with homebrew.

**Comment by fnando** [comment]
I won't have access to a Windows machine until the end of August, but I can take a look after that.

**Comment by leighmcculloch** [comment]
- https://github.com/Homebrew/homebrew-core/pull/180191

**Comment by leighmcculloch** [comment]
I striked the apt deb because I realised no one has asked for this, and recently we used homebrew in a docker image where we might have otherwise used the deb and it worked fine.

I striked the docker image for similar reasons, to pare this back for higher impact and reduced maintenance on lower impact things.

**Comment by Ifropc** [comment]
I agree that publishing in apt is low impact + not everyone (I think not even 50% of desktop users) is using apt + people _should be_ more comfortable installing it using cargo

**Comment by janewang** [comment]
Need legal clearance

**Comment by janewang** [comment]
Based on emails from legal, there's no need to disclose any 3rd party contributions in s3 as all contribution is original. This issue is not blocked on the legal front.

**Comment by fnando** [comment]
* pr adding a windows installer generator: https://github.com/stellar/stellar-cli/pull/1726
* pr releasing the initial version on winget: https://github.com/microsoft/winget-pkgs/pull/191259 (first release must be manual, then we can automate the process)
* ticket asking for PAT: https://github.com/stellar/ops/issues/3350
* ticket asking for `microsoft/winget-pkgs` repo fork: https://github.com/stellar/ops/issues/3351

**Comment by janewang** [comment]
Working on automation

**Comment by janewang** [comment]
Ops request update for PAT: https://github.com/stellar/ops/issues/3350

**Comment by leighmcculloch** [comment]
> * ticket asking for PAT: [Add PAT with scope `public_repo` to stellar-cli repository ops#3350](https://github.com/stellar/ops/issues/3350)

What is this PAT needed for?

**Comment by janewang** [comment]
@leighmcculloch can we drop winget and use wsl?

**Comment by leighmcculloch** [comment]
Those two things are not mutually exclusive. We can add support for the most common WSL case by releasing a Debian/Ubuntu compatible deb. @fnando suggested this [here](https://stellarfoundation.slack.com/archives/C06KTGUULUF/p1739936981595819?thread_ts=1739934747.951389&cid=C06KTGUULUF).

**Comment by janewang** [comment]
We are not going to do `cargo binstall`
